gem/templates/springster/comments/reply.html

```diff
@@ -1,31 +1,208 @@
+      {% endif %}
+        <span class="comments__publish-date">
+          {{node.submit_date|timesince}} {% trans "ago" %}
+        </span>
+      <div class="comments__reply-meta">
+          {% if not node.user|is_in_admin_group %}
+            <a href="{% url 'report_comment' node.pk %}" class="call-to-action__nav-item-text call-to-action__nav-item-text--report">
+              {% trans "Report" %}
+            </a>
+          {% endif %}
+          {% if node.parent == None %}
+            <a href="{% url 'molo.commenting:molo-comments-reply' node.pk %}#comment-block" class="call-to-action__nav-item-text call-to-action__nav-item-text--response">
+            {% trans "Reply"%}</a>
+            {% if do_not_link_replies or node.get_children.count == 0 %}
+              <p class="call-to-action__nav-item-text call-to-action__nav-item--replies-counter">
+                {{ node.get_children.count }}
+                {% if node.get_children.count == 0 or node.get_children.count == 1 %}
+                  {% trans "Reply" %}
+                {% else %}
+                  {% trans "Replies" %}
+                {% endif %}
+              </p>
+            {% else %}
+              <a href="{% url 'molo.commenting:molo-comments-reply' node.pk %}#comment-block" class="call-to-action__nav-item-text call-to-action__nav-item--replies-counter">
+                {{ node.get_children.count }}
+                {% if node.get_children.count == 0 or node.get_children.count == 1 %}
+                  {% trans "Reply" %}
+                {% else %}
+                  {% trans "Replies" %}
+                {% endif %}
+              </a>
+            {% endif %}
+        {% endif %}
+      </div>
+      {% endif %}
+      {% if node.parent == None and node.get_children.count != 0 %}
+      <h3 class="comment__replies-title">{% trans "Recent Replies"%}</h3>
+      {% endif %}
+    </div>
+  {% endfor %}
+  <div class="pagination">
+    {% if replies.has_previous %}
+      <a href="?p={{ replies.previous_page_number }}" class="pagination__nav-anchor-previous"></a>
+    {% endif %}
+    <span class="pagination__current-page">
+      {% trans "Page" %} {{ replies.number }} {% trans "of" %} {{ replies.paginator.num_pages }}
+    </span>
+    {% if replies.has_next %}
+      <a href="?p={{ replies.next_page_number }}" class="pagination__nav-anchor-next"></a>
     {% endif %}
-    {% for node in replies reversed %}
-      {% include "comments/comment.html" %}
-    {% endfor %}
-  {% include "patterns/basics/pagination/pagination-comments.html"%}
+  </div>
   <div id="comment-block">
     <div class="heading heading__component">
       <h2>{% trans "Add your reply" %}</h2>
     </div>
     {% if request.user.is_authenticated %}
-      {% include "comments/form.html" with node=comment %}
+      <form id="comment-form" action="{% url 'molo.commenting:molo-comments-post' %}#comments-list" method="post" class="comments__form" novalidate>
```

Comment:
@projectmushroom with the forms I do think we can reuse although we can reuse plenty of things and paying for it. Thoughts?

Reply:
mmm, I think its best if we try to collapse everything, If we don't, like you said, we can reuse plenty of things, and yes pay for it in the end.  😞